common/functions/src/scalars/geos/geo_to_h3.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+use common_datavalues::prelude::*;
+use common_datavalues::with_match_primitive_type_id;
+use common_datavalues::ColumnBuilder;
+use common_datavalues::ColumnRef;
+use common_datavalues::ColumnsWithField;
+use common_datavalues::DataTypeImpl;
+use common_datavalues::ScalarColumn;
+use common_datavalues::Series;
+use common_datavalues::TypeID;
+use common_exception::ErrorCode;
+use geo_types::Coordinate;
+use h3ron::H3Cell;
+use h3ron::Index;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionContext;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct GeoToH3Function {
+    display_name: String,
+}
+
+impl GeoToH3Function {
+    pub fn try_create(
+        display_name: &str,
+        args: &[&DataTypeImpl],
+    ) -> common_exception::Result<Box<dyn Function>> {
+        let mut arg = args[0];
+        if arg.data_type_id() != TypeID::Float64 {
```

Comment:
we can use `assert_numeric`

Reply:
The `geo_to_h3`'s first and second args must be `Float64` and the third arg must be `UInt8`, so I don't think `assert_numeric` is a good choice here.